internal/schema/typeref.go

```diff
@@ -75,7 +77,39 @@ func (r *TypeRef) GetTags() string {
 	return jsonTag + "\"`"
 }
 
-// GetName returns a recusive lookup of the type name
+// GetTypeNameWithOverride returns the typeName, taking into consideration any TypeOverride specified in the PackageConfig.
+func (r *TypeRef) GetTypeNameWithOverride(pkgConfig *config.PackageConfig) (string, error) {
+	var typeName string
+	var overrideType string
+	var err error
+
+	// Discover any TypeOverride override for the current field.
+	for _, p := range pkgConfig.Types {
+		if p.Name == r.GetName() {
+			if p.FieldTypeOverride != "" {
+				overrideType = p.FieldTypeOverride
+			}
+		}
+	}
+
+	// Set the typeName to the override or use what is specified in the schema.
+	if overrideType != "" {
+		typeName = overrideType
+	} else {
+		typeName, _, err = r.GetType()
+		if err != nil {
+			return "", err
+		}
+	}
+
+	if r.IsList() {
+		return fmt.Sprintf("[]%s", typeName), nil
```

Comment:
While perhaps handy, is this the right spot for this block? A type being a list/array/slice isn't part of it's _name_

Reply:
Okay, I'll relocate this.